packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,393 @@
+  testResultCount500Rate01,
+  testResultCount500Rate1
+} from './bloom_filter_golden_test_data';
+
+describe('BloomFilter', () => {
+  it('can initiate an empty bloom filter', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.size).to.equal(0);
+  });
+
+  it('empty bloom filter can have positive hash count', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 1);
+    expect(bloomFilter.size).to.equal(0);
+  });
+
+  it('should throw error if empty bloom filter inputs are invalid', () => {
+    try {
+      new BloomFilter(new Uint8Array(0), 1, 0);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: A valid empty bloom filter should have 0 padding.'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('can initiate a non empty bloom filter', () => {
+    const bloomFilter = new BloomFilter(
+      new Uint8Array([151, 153, 236, 116, 7]),
+      3,
+      13
+    );
+    expect(bloomFilter.size).to.equal(37);
+  });
+
+  it('should throw error if padding is negative', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), -1, 1);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Padding is negative.'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('should throw error if bitmap size is negative', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), 9, 1);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Bitmap size is negative.'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('should throw error if hash count is negative', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), 1, -1);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Hash count is 0 or negative'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('should throw error if hash count is 0 for non empty bloom filter', () => {
+    try {
+      new BloomFilter(new Uint8Array(1), 1, 0);
+      expect.fail();
+    } catch (error) {
+      expect(
+        (error as Error)?.message.includes(
+          'INTERNAL ASSERTION FAILED: Hash count is 0 or negative'
+        )
+      ).to.be.true;
+    }
+  });
+
+  it('mightContain in empty bloom filter should always return false', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.mightContain('abc')).to.be.false;
+    expect(bloomFilter.mightContain('def')).to.be.false;
+  });
+
+  it('mightContain should always return false for empty string', () => {
+    const emptyBloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    const nonEmptyBloomFilter = new BloomFilter(
+      new Uint8Array([151, 153, 236, 116, 7]),
```

Comment:
This looks like a very carefully constructed bloom filter; however, this test does not require such precision. Instead, use a bloom filter that has all bits set to guarantee that the empty string is not accidentally being hashed to values that are legitimately not present in the bloom filter.

Reply:
Bump